### PR TITLE
Allow custom export columns

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @NoahCarnahan here's my first go at allowing RemoteApps to use our new export system. This doesn't allow for any repeat groups/arrays in the data submitted. To allow for that there are two options I think:
1. Allow people to specify custom tables and the path to those tables (can get hairy with nested arrays...). This I think is more work and complexity.
2. If a question is an array just stringify the result and put it in the cell. This is doesn't allow for very good exports, but drastically reduces LOE and complexity.

Here is a screen shots of how this currently works:

screen shot 2016-07-24 at 10 24 37 am

:fish: or 🏠

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 12 errors:

* corehq/apps/export/static/export/js/models.js, line 164 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 170 - Missing semicolon. (semi)
* corehq/apps/export/static/export/js/models.js, line 178 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 276 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 278 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 284 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 286 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 290 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 294 - 'UserDefinedExportColumn' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 306 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/utils.js, line 34 - Missing semicolon. (semi)
* corehq/apps/export/static/export/spec/Exports.Utils.spec.js, line 35 - Missing semicolon. (semi)

**Comment by czue** [comment]
your proposal 1 sounds ok to me in the short term. will we be able to cleanly migrate these across from the old format?

**Comment by czue** [comment]
i'll let @NoahCarnahan handle the code review since he's got more context

**Comment by benrudolph** [comment]
yeah i'll think about the conversion. going with proposal 1 will allow the conversion to be more seamless (currently that conversion process hasn't been implemented for remote apps)

File: corehq/apps/export/static/export/js/utils.js [hunk 1]
```diff
@@ -19,9 +19,48 @@ hqDefine('export/js/utils.js', function () {
...
+    /**
+     * customPathToNodes
+     *
+     * This function takes a string path like form.meta.question and
+     * returns the equivalent path in an array of PathNodes.
+     *
+     * @param {string} customPathString - A string dot path to be converted
+     *      to PathNodes.
+     * @returns {Array} Returns an array of PathNodes.
+     */
```
Comment (NoahCarnahan) [inline corehq/apps/export/static/export/js/utils.js]: Yay docs!

**Comment by NoahCarnahan** [comment]
Nice, thanks Ben.

**Comment by benrudolph** [comment]
😀

**Pull request merged**